### ISSUE TITLE
Let `zip` accept array of observables

### ISSUE THREAD

**Issue by menelaos** [Reactive-Extensions/RxJS#853]
Would it be possible for `zip` to accept an array of observables?

~~~javascript
const obsA = Rx.Observable.fromArray([1,2,3]);
const obsB = Rx.Observable.fromArray([4,5,6]);
const obsC = Rx.Observable.fromArray([7,8,9]);

// Works as expected
Rx.Observable.zip(obsA, obsB, obsC, function(a, b, c) {
    return a + b + c;
}).subscribe(function(result) {
    console.log(result); // 12, 15, 18
});

// Does not work (Uncaught TypeError: Cannot read property 'apply' of undefined)
Rx.Observable.zip([obsA, obsB, obsC], function(a, b, c) {
    return a + b + c;
}).subscribe(function(result) {
    console.log(result);
});
~~~

Of course, one could do

~~~javascript
Rx.Observable.zip(...[obsA, obsB, obsC], function(a, b, c) {
    return a + b + c;
}).subscribe(function(result) {
    console.log(result); // 12, 15, 18
});
~~~

but letting `zip` accept an array would make it consistent with e.g.
`combineLatest`.

**Comment by menelaos** [comment]
Cf. https://gist.github.com/mattpodwysocki/1360003

**Comment by xgrommx** [comment]
@menelaos you can use like so:

Rx.Observable.zip.apply(null, [obsA, obsB, obsC], function(a, b, c) {
    return a + b + c;
}).subscribe(function(result) {
    console.log(result);
});

**Comment by menelaos** [comment]
@xgrommx Thanks, I'm just wondering whether there's a conscious design decision behind `combineLatest` accepting an array and `zip` not. The gist above shows that it was at least considered at some point.

**Comment by paulpdaniels** [comment]
I think it may just be an oversight, I think it would be valuable to have it work similar to `combineLatest`.

**Comment by mattpodwysocki** [comment]
Fixed as per 3.0.2 with a new operator of `zipIterable` to handle arrays

**Comment by menelaos** [comment]
Thanks for your quick response! I'm curious as to why you have decided to make this a new operator. Wouldn't it have been possible to do something like
```javascript
Array.isArray(args[0]) && (args = args[0]);
```
in `zip` itself?
